### PR TITLE
Allow mail_handler to find case insensitive keys

### PR DESCRIPTION
**Pull request by oliverguenther**
https://community.openproject.com/wp/32251

### CODE REVIEW

**Comment by ulferts** [comment]
While I agree that this feature is desirable I see problems because of our current uniqueness checks. 
There could be two Types/Versions/... with almost identical names except for differences in upcase/downcase. Referencing those becomes non deterministic now. While I do not want to say that this is a common use case, we should prevent such clashes. 

How about we allow the behaviour but check whether more than one entry matches the lowercased key. In those cases, we bail, but accept the change as long as it is deterministic.

**Comment by oliverguenther** [comment]
Statuses are actually already tested for lowercase. How about we simply add a validation to avoid case-differency-only types and versions as well? I'm fine with your approach for the course of this PR however

**Comment by ulferts** [comment]
Turning the validations case insensitive is fine with me @oliverguenther.

**Comment by oliverguenther** [comment]
I changed the validations to case-insensitive and created a Feature so it turns up in the changelog
https://community.openproject.com/wp/32513

I also changed the branch to dev to avoid merging this in a patch release

File: app/models/enumeration.rb [hunk 1]
```diff
@@ -40,9 +40,9 @@ class Enumeration < ActiveRecord::Base
   before_destroy :check_integrity
 
   validates_presence_of :name
-  validates_uniqueness_of :name, scope: %i(type project_id)
-  validates_length_of :name, maximum: 30
```
Comment (ulferts) [inline app/models/enumeration.rb:44]: If this is an intended change, we need to also change the type in the database.

**Review by ulferts — CHANGES_REQUESTED** [review]
Admittedly being paranoid here, but we probably need a migration for this change to alter formerly valid enumerations/categories/... that now became duplicates.

File: app/models/enumeration.rb (hunk 1 above)
Comment (oliverguenther) [reply app/models/enumeration.rb:44]: No this was intended, thanks for catching

**Review by ulferts — APPROVED** [review]
Looks good :+1:

We talked about the necessity of adding migrations and agreed to not need them.

**Pull request merged**